src/dos/dos_devices.cpp

```diff
@@ -547,17 +547,21 @@ bool DOS_DeviceHasName(const RealPt rp, const std::string_view req_name)
 	const auto segment    = RealSegment(rp);
 	const auto offset     = RealOffset(rp) + DeviceDriverInfo::name_offset;
 
-	const auto search_len = std::min(req_name.length(),
-	                                 DeviceDriverInfo::name_length);
-
-	for (uint8_t i = 0; i < search_len; ++i) {
-		// source and requested characters
-		const auto s = real_readb(segment, check_cast<uint16_t>(offset + i));
-		const auto r = req_name[i];
-		if (r == '\0') {
-			break;
+	const auto req_length = req_name.length();
+	if (req_length > DeviceDriverInfo::name_length) {
+		return false;
+	}
+
+	for (uint8_t i = 0; i < DeviceDriverInfo::name_length; ++i) {
+		const auto source = static_cast<char>(real_readb(segment, check_cast<uint16_t>(offset + i)));
+
+		// Ensure both strings have the same length.
+		// DOS name is not necessarily null terminated but padded by spaces.
+		if (i == req_length) {
+			return source <= 0x20;
```

Comment:
I've read this several time, and I'm still confused. I guess I could trace it with various inputs and figure it out, but this calls for clarification & simplification:

1. The comment is misleading as it implies null and spaces are used for termination. But we're checking for `<= 0x20`. So any control char is ok? Should we not be exact then and only check for 0x00 and 0x20?
2. A bit confused what happens if `req_length < DeviceDriverInfo::name_length`.
3. Can't this be moved outside of the loop? If would feel more natural and less confusing that way. `i == req_length` can only happen once the for loop is over.

Overall, I'd like to have a simpler implementation so I can see at a glance what's going on.

Reply:
The ‘source <= 0x20’ was taken from old code before the refactor that introduced this bug.  If you know what the DOS spec says I can change it.  Or I can try to look it up later.

To your other points, yeah I could read the DOS name into a ‘std::string’ up front and probably that would be easier to read.

Reply:
> If you know what the DOS spec says I can change it. Or I can try to look it up later.

I don't know, just reading this for the first time it raise a lot of questions, that's all 😄 Probably safest to adjust the comment to the code then.

Reply:
Done.  This should be easier to read if you can live with the performance penalty of copying 8 bytes into a `std::string` :wink:

Reply:
Very nice fix, @weirddan455 !

Reply:
Thanks @weirddan455. Now the intent isn't clouded by weird string micromanagement stuff. 🎉 The "performance penalty" is about on the same level as the "fuel consumption penalty" when driving the car with your hats on 😎